src/datagrid/DataGrid.tsx

```diff
@@ -312,6 +293,76 @@ export class DataGrid extends React.PureComponent<DataGridProps, DataGridState>
         }
     }
 
+    // funtion to initialize datagrid state
+    initializeDataGridState = (): DataGridState => {
+        const {isLoading, itemText} = this.props;
+        const rows = this.initializeRowData();
+        const columns = this.initializeColumnData();
+        const state: DataGridState = {
+            isLoading: isLoading || false,
+            selectAll: this.isAllRowsSelected(rows),
+            allColumns: [...columns],
+            allRows: [...rows],
+            itemText: itemText || DEFAULT_ITEM_TEXT,
+            pagination: this.initializePaginationData(),
+        };
+        return state;
+    };
+
+    // Function to initialize rows data in state
+    initializeRowData = (): DataGridRow[] => {
+        const {rows} = this.props;
+        let updatedRows: DataGridRow[] = [];
+        if (rows && rows.length) {
+            updatedRows = this.updateRowIDs(rows);
+            updatedRows.forEach(function(row) {
+                const rowSelectionIsDisabled = row.disableRowSelection !== undefined ? row.disableRowSelection : false;
```

Comment:
**row.disableRowSelection !== undefined  ? ** 
can be replaced with 
**row.disableRowSelection ?**  as well.

Reply:
@maheshkulkarni4 `disableRowSelection `is a `Boolean `field So here I wanted to check if user has passed it or not.